src/psfmachine/utils.py

```diff
@@ -89,6 +90,77 @@ def get_gaia_sources(ras, decs, rads, magnitude_limit=18, epoch=2020, dr=2):
     return gd.data.to_pandas()
 
 
+def do_tiled_query(
+    ra_grid, dec_grid, ngrid=(5, 5), magnitude_limit=18, epoch=2020, dr=3
+):
+    """
+    Find the centers and radius of the query when the sky area is large. This function
+    divides the data into `ngrid` tiles and compute the ra, dec coordinates for each
+    tile as well as its radius.
+
+    Parameters
+    ----------
+    ra_grid : numpy.ndarray
+        Data array with values of Right Ascension.
+    dec_grid : numpy.ndarray
+        Data array with values of Declination.
+    ngrid : tuple
+        Tuple with number of bins in each axis. Default is (5, 5).
+    magnitude_limit : int
+        Limiting magnitued for query
+    epoch : float
+        Year of the observation (Julian year) used for proper motion correction.
+    dr : int
+        Gaia Data Release to be used, DR2 or EDR3.
+
+    Returns
+    -------
+    sources : pandas.DatFrame
+        Pandas DatFrame with number of result sources (rows) and Gaia columns
+    """
+    # find edges of the bins
+    ra_edges = np.histogram_bin_edges(ra_grid, ngrid[0])
+    dec_edges = np.histogram_bin_edges(dec_grid, ngrid[1])
+    sources = []
+    # iterate over 2d bins
+    for ix in range(1, len(ra_edges)):
+        for jd in range(1, len(dec_edges)):
+            # check if image data fall in the bin
+            _in = (
+                (ra_edges[ix - 1] <= ra_grid)
+                & (ra_grid <= ra_edges[ix])
+                & (dec_edges[jd - 1] <= dec_grid)
+                & (dec_grid <= dec_edges[jd])
+            )
+            if not _in.any():
+                continue
+            # get the center coord of the query and radius
+            ra_in = ra_grid[_in]
+            dec_in = dec_grid[_in]
+            ra_q = ra_in.mean()
+            dec_q = dec_in.mean()
+            rad_q = np.hypot(ra_in - ra_q, dec_in - dec_q).max() + 10 / 3600
```

Comment:
probably similar problem here

Reply:
add a test for an ra patch that goes over 0-360

Reply:
une tests added